### PR TITLE
7039014: Confusing error message for method conflict

### PR DESCRIPTION
**Pull request by archiecobbs**
This fixes a confusing error message.

Currently, this input:
```java
interface A<T> {
    byte m(String x);
    char m(T x);
}

interface B extends A<String> {
}
```
produces this error:
```
A.java:6: error: types A<String> and A<String> are incompatible;
interface B extends A<String> {
^
  both define m(String), but with unrelated return types
1 error
```
This patch detects when the two types are actually the same type and gives this error message instead:
```
A.java:6: error: type A<String> defines m(String) more than once with unrelated return types
interface B extends A<String> {
^
1 error
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-7039014](https://bugs.openjdk.org/browse/JDK-7039014): Confusing error message for method conflict

### Reviewers
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10752/head:pull/10752` \
`$ git checkout pull/10752`

Update a local copy of the PR: \
`$ git checkout pull/10752` \
`$ git pull https://git.openjdk.org/jdk pull/10752/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10752`

View PR using the GUI difftool: \
`$ git pr show -t 10752`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10752.diff">https://git.openjdk.org/jdk/pull/10752.diff</a>

</details>

### CODE REVIEW

File: test/langtools/tools/javac/Diagnostics/7039014/T7039014a.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+ * @test /nodynamiccopyright/
+ * @bug     7039014
+ * @summary Confusing error message for method conflict
+ * @author archiecobbs
```
Comment (vicente-romero-oracle) [inline test/langtools/tools/javac/Diagnostics/7039014/T7039014a.java:5]: nit: we don't add the `@author` tag anymore, it has been kept only if the original test had it, usually very old tests.

File: test/langtools/tools/javac/Diagnostics/7039014/T7039014a.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * @compile/fail/ref=T7039014a.out -XDrawDiagnostics T7039014a.java
+ */
+public class T7039014a {
```
Comment (vicente-romero-oracle) [inline test/langtools/tools/javac/Diagnostics/7039014/T7039014a.java:9]: I think that T7039014(a,b,c) should be combined into only one test

File: test/langtools/tools/javac/Diagnostics/7039014/T7039014a.java (hunk 1 above)
Comment (archiecobbs) [reply test/langtools/tools/javac/Diagnostics/7039014/T7039014a.java]: Thanks - removed.

File: test/langtools/tools/javac/Diagnostics/7039014/T7039014a.java (hunk 2 above)
Comment (archiecobbs) [reply test/langtools/tools/javac/Diagnostics/7039014/T7039014a.java]: Good idea, thanks - done.

**Review by vicente-romero-oracle — APPROVED** [review]
looks good

**Pull request closed without merging**